opdracht4.js

```diff
@@ -0,0 +1,193 @@
+
+//Opdracht 4c: Zorg ervoor dat er een string wordt gegenereerd voor alle beschikbare schermgroottes van één tv, in zowel
+// inches als cm. Doe dit door een functie te schrijven die één screen-sizes array verwacht ( zoals
+// inventory[0].availableSizes) en de groottes op de volgende manier samenvoegt: [schermgrootte] inches
+// ([schermgrootte omgerekend]cm) | [schermgrootte] inches ([schermgrootte omgerekend]cm) etc.
+// Als een tv maar één schermgrootte heeft ([32]) wordt de output 32 inch (81 cm). Wanneer een tv vier schermgroottes
+// heeft ([43, 50, 55, 58]) wordt de output 43 inch (109 cm) | 50 inch (127 cm) | 58 inch (147 cm). Test of jouw functie
+// werkt in alle mogelijke gevallen.
+
+function screenSize(arrayOfPrizes) {
+    const inchAndCM = arrayOfPrizes.map((screenSize) => {
+        const inchSize = screenSize;
+        const cmSize = Math.round(screenSize * 2.54);
+        return `${inchSize} inch (${cmSize} cm)`;
+    });
+    return inchAndCM.join(" | ");
+}
+console.log(screenSize(inventory[0].availableSizes));
+
+//Opdracht 4d: Zorg ervoor de informatie van één van de tv's zoals het voorbeeld wordt weergegeven op de pagina.
+// Gebruik hiervoor de functies die je hebt gemaakt in opdracht 4a, 4b en 4c.
+function letsDoItAll(arrayItem){
+    return `${tvName(arrayItem)}\n${tvPrice(arrayItem.price)}\n${screenSize(arrayItem.availableSizes)}`
+}
+console.log(letsDoItAll(inventory[4]));
+
+//Opdracht 4e: Schrijf een functie die ALLE tv's weergeeft op de pagina zoals in het voorbeeld. Dit wil je natuurlijk
+// niet acht keer opnieuw schrijven, want nu zijn het 8 tv's, maar in de toekomst misschien wel 200!
+// Gebruik in deze functie de voorgaande functies die je hebt geschreven, om onderdelen van de data te formatten.
+// De overkoepelende "tv-generator-functie" verwacht één parameter: de volledige array met tv-objecten. Vergeet 'm niet aan te roepen!
+
+inventory.map((tvItem) =>  console.log(letsDoItAll(tvItem)));
+
+//Bonus opdracht
+//1. Maak drie knoppen op de pagina: `Sorteer op prijs`, `AmbiLight TV's` en `Uitverkochte exemplaren`. Gebruik de code
+//die je in opdracht 1b, 1c en 1d hebt gemaakt en schrijf dit om naar functies zodat je ze kunt aanroepen op het moment
+//dat de buttons geklikt worden. Zorg ervoor dat de functies de uitkomsten in de console loggen als de gebruiker op
+//de bijbehorende knop klikt. _Tip_: lees hiervoor paragraaf 7.4 op EdHub eens door!
+//maak knoppen
+//schrijf functies om de code in opdracht 1b, c en d
+//koppel deze functies aan de knoppen
+//zorg dat bij een button click antwoorden gelogt worden
+
+
+function findSoldOut(){
+    let soldOutList = inventory.filter((tvObject)=> {
+        if(tvObject.originalStock - tvObject.sold === 0){
+            return tvObject;
+        }
+
+    });
+    console.log(soldOutList);
+}
+
+
+function findAmbilight(){
+    const ambiLight = inventory.filter((tvObject)=>{
+        return tvObject.options.ambiLight === true;
```

Comment:
Omdat `ambiLight` al een boolean oplevert, hoef je dit niet expliciet te controleren:
```suggestion
        return tvObject.options.ambiLight;
```